tests/e2e/csi_cns_telemetry.go

```diff
@@ -0,0 +1,278 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+)
+
+var _ = ginkgo.Describe("[csi-vanilla] CNS-CSI Cluster Distribution Telemetry", func() {
+	f := framework.NewDefaultFramework("csi-cns-telemetry")
+	var (
+		client       clientset.Interface
+		namespace    string
+		scParameters map[string]string
+		datastoreURL string
+	)
+
+	ginkgo.BeforeEach(func() {
+		bootstrap()
+		client = f.ClientSet
+		namespace = getNamespaceToRunTests(f)
+		scParameters = make(map[string]string)
+		datastoreURL = GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		csivsphereConf := "csi-vsphere.conf"
+		// Fetch existing vsphere-config-secret
+		currentSecret, getSecretErr := client.CoreV1().Secrets(kubeSystemNamespace).Get(ctx, configSecret, metav1.GetOptions{})
+		gomega.Expect(getSecretErr).NotTo(gomega.HaveOccurred())
+		framework.Logf("Secret Name is %s\n", currentSecret.Name)
+
+		// Read and map the secret values to cfg
+		originalConf := string(currentSecret.Data[csivsphereConf])
+		cfg, readerr := readConfigFromSecretString(originalConf)
+		gomega.Expect(readerr).NotTo(gomega.HaveOccurred())
+
+		framework.Logf("Check Cluster-distribution value before running the test %s\n", cfg.Global.ClusterDistribution)
+
+		// Check if the cluster-distribution value is CSI-Vanilla or not, if not reset
+		if cfg.Global.ClusterDistribution != "CSI-Vanilla" {
```

Comment:
You are using "CSI-Vanilla" in multiple places, declare it under a variable in e2e_common.go